lib/topological_inventory/ansible_tower/receptor/response_worker.rb

```diff
@@ -0,0 +1,73 @@
+module TopologicalInventory::AnsibleTower
+  module Receptor
+    class ResponseWorker
+      attr_reader :started
+      alias_method :started?, :started
+
+      def initialize
+        self.started = false
+        @registered_messages = {}
+      end
+
+      def start
+        return if started
+
+        self.started = true
+        Thread.new { listen }
+      end
+
+      def register_msg_id(msg_id, api_object, response_method = :response_received)
+        @registered_messages[msg_id] = { :api_object => api_object, :method => response_method }
```

Comment:
This isn't guaranteed to be thread safe if you're using this to register from one thread while reading from it [here](https://github.com/RedHatInsights/topological_inventory-ansible_tower/pull/46/files#diff-2fde49ec8f16e8d8a0038a1e85c8dae3R44)

You could use `Concurrent::Map`

Reply:
good point